src/main/java/baseball/domain/GameChoice.java

```diff
@@ -0,0 +1,30 @@
+package baseball.domain;
+
+import static baseball.handler.ErrorHandler.INVALID_NUMBER;
+
+public class GameChoice {
+
+    private final int choice;
+    private final int RESTART_GAME = 1;
+    private final int FINISH_GAME = 2;
+
+    public GameChoice() {
+        this.choice = RESTART_GAME;
+    }
+
+    public GameChoice(int choice) {
+        validate(choice);
+
+        this.choice = choice;
+    }
```

Comment:
GameChoice의 생성자에 따라 결과가 달라지는 것 같아요.
인자값 없이 사용되면, RESTART_GAME을 선언하고,
인자값이 있다면 choice를 검증 후 반영하는 생성자인 것 같아요.

생성자 오버로딩으로 설계할 때는 정적 팩토리 메소드를 고려해보시는 것도 좋을 것 같습니다.

```suggestion
    private GameChoice() {
        this.choice = RESTART_GAME;
    }

    private GameChoice(int choice) {
        validate(choice);

        this.choice = choice;
    }
    
   public static GameChoice restart(){
       return new GameChoice();
   } 
    
   public static GameChoice ????(int choice){
       return new GameChoice(choice);
   } 
```

이런 식으로 설계한다면 생성자를 오버로딩해서 설계해도, 누구나 해당 메소드가 어떤 의미로 어디에 사용되어야 할 지 직관적으로 파악이 가능할 것 같네요!

Reply:
이 답변은 진짜 놀랐어요😲!! 생각하지 못했던 방향성인데 생성자를 오버로딩하는 경우가 있다면 이런식으로 사용하는게 훨씬 직관적일 것 같습니다!